pkg/asset/quota/openstack/openstack.go

```diff
@@ -25,14 +43,36 @@ func Constraints(ci *validation.CloudInfo, controlPlanes []machineapi.Machine, c
 	var ret []quota.Constraint
 	ret = controlPlane(ci, ctrplConfigs)
 	ret = append(ret, compute(ci, computeReplicas, computeConfigs)...)
+	ret = append(ret, networkConstraints(networkType)...)
 	return aggregate(ret)
 }
 
+func networkConstraints(networkType string) []quota.Constraint {
+	var ret []quota.Constraint
+	ret = append(ret, generateConstraint("Router", int64(routersConstraint)))
+	if networkType == "kuryr" {
+		ret = append(ret, generateNetworkConstraints(int64(portsConstraintWithKuryr), int64(subnetsConstraintWithKuryr), int64(networksConstraintWithKuryr), int64(securityGroupsConstraintWithKuryr), int64(securityGroupRulesConstraintWithKuryr))...)
+	} else {
+		ret = append(ret, generateNetworkConstraints(int64(portsConstraintKuryr), int64(subnetsConstraintKuryr), int64(networksConstraintKuryr), int64(securityGroupsConstraintKuryr), int64(securityGroupRulesConstraintKuryr))...)
+	}
+	return ret
+}
+
+func generateNetworkConstraints(port int64, subnet int64, network int64, securityGroup int64, securityGroupRule int64) []quota.Constraint {
+	var ret []quota.Constraint
+	ret = append(ret, generateConstraint("Port", port))
+	ret = append(ret, generateConstraint("Subnet", subnet))
+	ret = append(ret, generateConstraint("Network", network))
+	ret = append(ret, generateConstraint("SecurityGroup", securityGroup))
+	return append(ret, generateConstraint("SecurityGroupRule", securityGroupRule))
+}
+
 func controlPlane(ci *validation.CloudInfo, machines []*openstackprovider.OpenstackProviderSpec) []quota.Constraint {
 	var ret []quota.Constraint
 	for _, m := range machines {
 		flavorInfo := ci.Flavors[m.Flavor]
 		ret = append(ret, machineFlavorCoresToQuota(flavorInfo, ci), machineFlavorRAMToQuota(flavorInfo, ci))
+		ret = append(ret, generateConstraint("Port", int64(len(m.Networks))))
```

Comment:
Question (on the previous line 74, really): can it be that if you have 3 ControlPlane Machines defined with 16GB RAM each, then you are setting a "16GB" requirement three times, instead of setting a requirement of 16*3GB?

Reply:
I think this is the safest thing to do here as there are 3 separate Machine profiles. It may seem safe to assume that they're all the same, but the performance penalty of handling them separately is inconsequential. This is simple and avoids an assumption.

Reply:
I agree with @mdbooth and it was my reasoning when we implementing this.

Reply:
I agree with the no-assumptions rationale. My concern was that we weren't adding up the RAM, disk, and vCPU requirements (I am commenting code that is not touched in this PR).

Reply:
@pierreprinetti to follow-up on our conversation, today I confirmed that the existing code here works fine.
I reduced my quotas on purpose to see what error message I would get when trying to deploy 3 master with a flavor of 4 Cores and 16GB of RAM + 3 workers with a flavor of 4 Cores and 8GB of RAM:

```FATAL failed to fetch Cluster: failed to fetch dependency of "Cluster": failed to generate asset "Platform Quota Check": error(MissingQuota): Cores is not available because the required number of resources (24) is more than the limit of 2```

The required number of resources is right (3x4 + 3x4) for the Cores.
Same for RAM:

```FATAL failed to fetch Cluster: failed to fetch dependency of "Cluster": failed to generate asset "Platform Quota Check": error(MissingQuota): RAM is not available because the required number of resources (73728) is more than the limit of 2```

3 x 16 + 3 x 8 = 72GB

Reply:
Perfect! Thanks